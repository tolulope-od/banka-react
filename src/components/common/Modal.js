import React from 'react';
import PropTypes from 'prop-types';

const Modal = ({ onClick, modalContent, className }) => {
  return (
    <div id="myModal" className={className} onClick={onClick}>
      <div className="modal-content" id="account-modal-content" onClick={onClick}>
        <span className="close" onClick={onClick} id="close-button">
          &times;
        </span>
        <div>{modalContent}</div>
      </div>
    </div>
  );
};

Modal.propTypes = {
  onClick: PropTypes.func.isRequired,
  modalContent: PropTypes.object.isRequired,
  className: PropTypes.string.isRequired
};

export default Modal;
