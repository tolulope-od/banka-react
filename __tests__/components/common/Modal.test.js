import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Modal from '../../../src/components/common/Modal';

const shallowSetup = () => {
  const props = {
    onClick: () => ({}),
    modalContent: {},
    className: 'modal'
  };

  const enzymeWrapper = shallow(<Modal {...props} />);

  return { props, enzymeWrapper };
};

describe('<Modal /> Component', () => {
  const { props } = shallowSetup();

  it('Renders the Modal component', () => {
    const wrapper = shallow(<Modal {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
