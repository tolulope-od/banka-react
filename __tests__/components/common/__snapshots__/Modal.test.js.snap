// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Component Renders the Modal component 1`] = `
<div
  className="modal"
  id="myModal"
  onClick={[Function]}
>
  <div
    className="modal-content"
    id="account-modal-content"
    onClick={[Function]}
  >
    <span
      className="close"
      id="close-button"
      onClick={[Function]}
    >
      Ã—
    </span>
    <div>
      <Component />
    </div>
  </div>
</div>
`;
